doc/tutorial/single_cell_detailed_recipe.rst

```diff
@@ -49,71 +49,78 @@ examine the recipe in detail: how to create one, and why it is needed.
-the *hh* and *pas* mechanisms but not the *Ih* mechanism, which is present in the allen catalogue. To be able
-to use *Ih*, we extend ``self.the_cat`` to include the allen catalogue.
+   1. the recipe constructor explicitly calls the base C++ class constructor (at the moment there is no way to
+   instruct Python to do that automatically).
+
+   2. the mechanism catalogue is assigned to an instance variable, such that its lifetime is guaranteed to extend
+   to the entire duration of the simulation.
+   
+
+Step **(5.1)** defines the class constructor. Before variable initialization, we call the base C++ class 
+constructor ``arbor.recipe.__init__(self)``. This ensures correct initialization of memory in the C++ class.
+
+We also create the instance variable for the mechanism catalogue, the ``self.the_cat`` variable. We set it to
+Arbor's default mechanism catalogue. This will expose the *hh* and *pas* mechanisms but not the *Ih* mechanism,
+which is present in the allen catalogue. To be able to use *Ih*, we extend ``self.the_cat`` to include the Allen
+catalogue.
 
 Finally we create the ``self.the_props`` variable. This will hold the global properties of the model, which apply
 to all the cells in the network. Initially it is empty. We set all the properties of the system similar to
 what we did in the :ref:`previous example <tutorialsinglecellswc-gprop>`. One last important step is to register
 ``self.the_cat`` with ``self.the_props``.
 
-.. Note::
-
-   The mechanism catalogue needs to live in the recipe as an instance variable. Its lifetime needs to extend
-   to the entire duration of the simulation.
-
-Step **(5.2)** overrides the :meth:`arbor.recipe.num_cells` method. It takes no arguments. We simply return 1,
+Step **(5.2)** overrides the :meth:`~arbor.recipe.num_cells` method. It takes no arguments. We simply return 1,
 as we are only simulating one cell in this example.
 
-Step **(5.3)** overrides the :meth:`arbor.recipe.cell_kind` method. It takes one argument: ``gid``.
+Step **(5.3)** overrides the :meth:`~arbor.recipe.cell_kind` method. It takes one argument: ``gid``.
 Given the gid, this method returns the kind of the cell. Our defined cell is a
 :class:`arbor.cell_kind.cable`, so we simply return that.
 
-Step **(5.4)** overrides the :meth:`arbor.recipe.cell_description` method. It takes one argument: ``gid``.
+Step **(5.4)** overrides the :meth:`~arbor.recipe.cell_description` method. It takes one argument: ``gid``.
 Given the gid, this method returns the cell description which is the cell object passed to the constructor
-of the recipe. We return ``self.the_cell``.
+of the recipe. We return ``cell``, the cell created just above.
 
-Step **(5.5)** overrides the :meth:`arbor.recipe.get_probes` method. It takes one argument: ``gid``.
+Step **(5.5)** overrides the :meth:`~arbor.recipe.get_probes` method. It takes one argument: ``gid``.
```

Comment:
Why the `~`?

Reply:
Tilde compression: to avoid printing `arbor.recipe.` all the time.